ebml/EbmlMaster.h

```diff
@@ -43,11 +43,6 @@
 #include "EbmlElement.h"
 #include "EbmlCrc32.h"
 
-#define EBML_MASTER_ITERATOR  std::vector<EbmlElement *>::iterator
-#define EBML_MASTER_CONST_ITERATOR  std::vector<EbmlElement *>::const_iterator
-#define EBML_MASTER_RITERATOR std::vector<EbmlElement *>::reverse_iterator
-#define EBML_MASTER_CONST_RITERATOR std::vector<EbmlElement *>::const_reverse_iterator
-
```

Comment:
These `#define`s are meant for the users of the library, not the library itself. Therefore changing them this way doesn't really achieve anything useful: it breaks the API by changing their name (from e.g. `EBML_MASTER_ITERATOR` to `libebml::EbmlMaster::EBML_MASTER_ITERATOR`) without providing any kind of benefit.

I'd be more OK with simply removing them. They're from a time before C++11 & `auto` were a thing & using iterators involved a lot more typing that it does nowadays.

Reply:
right. they seem to be used internally. Hence why I moved them into the namespace. SonarLint's complaint was that these defines should be typedef or using. Not possible outside the namespace.